client/environment.go

```diff
@@ -109,6 +110,42 @@ type EnvironmentCreate struct {
 	VcsCommandsAlias            string                `json:"vcsCommandsAlias"`
 }
 
+// When converted to JSON needs to be flattened. See custom MarshalJSON below.
+type EnvironmentCreateWithoutTemplate struct {
+	EnvironmentCreate EnvironmentCreate
+	TemplateCreate    TemplateCreatePayload
+}
+
+// The custom marshalJSON is used to return a flat JSON.
+func (create *EnvironmentCreateWithoutTemplate) MarshalJSON() ([]byte, error) {
```

Comment:
Since I wanted the support as explained above, I require a custom JSON marshal that merged the two structs to a single JSON.

Reply:
embedding those like [this](https://stackoverflow.com/a/24642709) won't flatten this out? (instead of the marshel -> unmarshel flatten) or maybe this is a restriction by the env without template struct you receive?

Reply:
It will not work because of some conflicts between the fields.
E.g. "Name".
In case of a conflict golang just ignores the field. (Therefore I would need to create a custom MarshalJSON any how).
I found this a little nicer and intuitive (The struct I mean).